### PR TITLE
Reject all uses of non-typaram bounds

### PR DESCRIPTION
**Pull request by jroesch**
This is a follow up to my recent PR. r? @nikomatsakis

### CODE REVIEW

**Comment by frewsxcv** [comment]
For reference, here's the other PR @jroesch mentioned: #20623

File: src/librustc_typeck/check/wf.rs [hunk 1]
```diff
@@ -267,6 +286,24 @@ impl<'ccx, 'tcx, 'v> Visitor<'v> for CheckTypeWellFormedVisitor<'ccx, 'tcx> {
...
+                match ty::impl_or_trait_item(self.ccx.tcx, local_def(method.id)) {
+                    ty::ImplOrTraitItem::MethodTraitItem(ty_method) => {
+                        reject_non_type_param_bounds(
+                            self.ccx.tcx,
+                            method.span,
+                            &ty_method.generics)
+                    }
+                    _ => {}
+                }
+            }
```
Comment (huonw) [inline src/librustc_typeck/check/wf.rs]: Shouldn't this have a `walk_trait_item` call somewhere?

Comment (nikomatsakis) [inline src/librustc_typeck/check/wf.rs]: Hmm, good point.

**Comment by alexcrichton** [comment]
Merged in #20733

**Pull request closed without merging**